### ISSUE TITLE
Getting "ERROR: no driver was registered for the schema " on scaffold built model

### ISSUE THREAD

**Issue by bjornforsberg** [kemalyst/kemalyst#32]
This continues on from #30 where I am starting a new app using the Readme guide. I have made no manual changes to the app itself, the only change compared to the Readme is that I called my model Shop instead of Post.

After migrating the database:

```
$ kgen migrate up
Migrating db, current version: 0, target: 20170331131409
OK   20170331131409_create_shop.sql
```

I rebuild the app successfully, and go to: `http://0.0.0.0:3000/shops`, I get the following 500 error:

```
ERROR: no driver was registered for the schema "", did you maybe forget to require the database driver? (ArgumentError)
0x10369c775: *CallStack::unwind:Array(Pointer(Void)) at ??
0x10369c711: *CallStack#initialize:Array(Pointer(Void)) at ??
0x10369c6e8: *CallStack::new:CallStack at ??
0x10367b811: *raise<ArgumentError>:NoReturn at ??
0x1037a3ef0: *DB::driver_class<(String | Nil)>:DB::Driver+:Class at ??
0x1037a3e3d: *DB::build_driver<URI>:DB::Driver+ at ??
0x1037a3e11: *DB::build_database<URI>:DB::Database at ??
0x1037a3df1: *DB::build_database<String>:DB::Database at ??
0x1037a3dd9: *DB::open<String>:DB::Database at ??
0x1037a5e26: *Shop@Kemalyst::Model::all<String, Array(Bool | Float32 | Float64 | Int32 | Int64 | Slice(UInt8) | String | Time | Nil)>:Array(Shop) at ??
0x1037a5aee: *Shop@Kemalyst::Model::all:Array(Shop) at ??
0x1037b629e: *ShopController::Index#call<HTTP::Server::Context>:String at ??
0x1037b11f7: *Kemalyst::Handler::Router#process_request<HTTP::Server::Context>:HTTP::Server::Context at ??
0x1037b0737: *Kemalyst::Handler::Router#call<HTTP::Server::Context>:HTTP::Server::Context at ??
0x1037d2fa2: *Kemalyst::Handler::Params@HTTP::Handler#call_next<HTTP::Server::Context>:(Bool | File::PReader | HTTP::Server::Context | HTTP::Server::Response | HTTP::Server::Response::Output | IO::FileDescriptor+ | Int32 | String | Nil) at ??
0x1037d159b: *Kemalyst::Handler::Params#call<HTTP::Server::Context>:(Bool | File::PReader | HTTP::Server::Context | HTTP::Server::Response | HTTP::Server::Response::Output | IO::FileDescriptor+ | Int32 | String | Nil) at ??
0x1037d10ea: *Kemalyst::Handler::Flash@HTTP::Handler#call_next<HTTP::Server::Context>:(Bool | File::PReader | HTTP::Server::Context | HTTP::Server::Response | HTTP::Server::Response::Output | IO::FileDescriptor+ | Int32 | String | Nil) at ??
0x1037d05cc: *Kemalyst::Handler::Flash#call<HTTP::Server::Context>:HTTP::Server::Context at ??
0x1037cd21a: *Kemalyst::Handler::Session@HTTP::Handler#call_next<HTTP::Server::Context>:(Bool | File::PReader | HTTP::Server::Context | HTTP::Server::Response | HTTP::Server::Response::Output | IO::FileDescriptor+ | Int32 | String | Nil) at ??
0x1037cc5bc: *Kemalyst::Handler::Session#call<HTTP::Server::Context>:HTTP::Server::Context at ??
0x1037ce248: *Kemalyst::Handler::Static@HTTP::Handler#call_next<HTTP::Server::Context>:(Bool | File::PReader | HTTP::Server::Context | HTTP::Server::Response | HTTP::Server::Response::Output | IO::FileDescriptor+ | Int32 | String | Nil) at ??
0x1037cda38: *Kemalyst::Handler::Static#call<HTTP::Server::Context>:(Bool | File::PReader | HTTP::Server::Context | HTTP::Server::Response | HTTP::Server::Response::Output | IO::FileDescriptor+ | Int32 | String | Nil) at ??
0x1037cff32: *Kemalyst::Handler::Error@HTTP::Handler#call_next<HTTP::Server::Context>:(Bool | File::PReader | HTTP::Server::Context | HTTP::Server::Response | HTTP::Server::Response::Output | IO::FileDescriptor+ | Int32 | String | Nil) at ??
0x1037cf6a0: *Kemalyst::Handler::Error#call<HTTP::Server::Context>:(Bool | File::PReader | HTTP::Server::Context | HTTP::Server::Response | HTTP::Server::Response::Output | IO::FileDescriptor+ | Int32 | String | Nil) at ??
0x1037cf162: *Kemalyst::Handler::Logger@HTTP::Handler#call_next<HTTP::Server::Context>:(Bool | File::PReader | HTTP::Server::Context | HTTP::Server::Response | HTTP::Server::Response::Output | IO::FileDescriptor+ | Int32 | String | Nil) at ??
0x1037ce911: *Kemalyst::Handler::Logger#call<HTTP::Server::Context>:HTTP::Server::Context at ??
0x1037dbfbd: *HTTP::Server::RequestProcessor#process<(OpenSSL::SSL::Socket::Server | TCPSocket+), (OpenSSL::SSL::Socket::Server | TCPSocket+), IO::FileDescriptor>:Nil at ??
0x1037db8d5: *HTTP::Server::RequestProcessor#process<(OpenSSL::SSL::Socket::Server | TCPSocket+), (OpenSSL::SSL::Socket::Server | TCPSocket+)>:Nil at ??
0x1037d3986: *HTTP::Server#handle_client<(TCPSocket+ | Nil)>:Nil at ??
0x1036971b6: ~procProc(Nil)@/usr/local/Cellar/crystal-lang/0.21.1_1/src/http/server.cr:163 at ??
0x1036a7f44: *Fiber#run:(IO::FileDescriptor | Nil) at ??
0x10368ce19: ~proc2Proc(Fiber, (IO::FileDescriptor | Nil))@/usr/local/Cellar/crystal-lang/0.21.1_1/src/fiber.cr:29 at ??
```

Which does not seem to be picking the adapter set in the model:

````
require "kemalyst-model/adapter/pg"

class Shop < Kemalyst::Model
  adapter pg

  # id, created_at and updated_at columns are automatically created for you.
  sql_mapping({
    url: String,
    token: String,
  })

end
````

Visiting `http://0.0.0.0:3000/` works fine.

Sorry to be sending over these issues, but more than happy to help with what I can (which is not much at this stage). However, the more people that can get this up and running easily, the better chance of wide adoption 😃

**Comment by bjornforsberg** [comment]
Ps. I have tried using the provided sample applications to try and work out where the issue lies here, but could not see any difference.

**Comment by drujensen** [comment]
can you confirm the DATABASE_URL is set?  If that doesn't work, can you try setting the driver directly in the config/database.yml?  For some reason its not finding the setting.

**Comment by bjornforsberg** [comment]
It was set correctly:

````
$ export DATABASE_URL=postgres://FORSBERGtwo:@localhost:5432/orderlylinks
$ echo $DATABASE_URL
postgres://FORSBERGtwo:@localhost:5432/orderlylinks
`````

However, when I hardcoded it into the `database.yml` file then it worked nicely. Strange thing is if I remove it again from the file, everything continues to work.

Closing this as most likely my own fault. Thanks again for your quick help!

Ps. Is there a reason the `database.yml` file no longer follows the Rails convention of specifying all the details (username, password, database name etc) in the file directly, and then just .gitignore it? I think that would help people moving from rails understand that step a little easier.

Happy to submit another PR if you are open to it.

Thanks again!
